### PR TITLE
Empty universe

### PR DESCRIPTION
**Pull request by lilyminium**
Implements #2383

Changes made in this Pull Request:
 - added empty Universe with 0 atoms

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/core/topology.py [hunk 1]
```diff
@@ -113,6 +113,9 @@ def make_downshift_arrays(upshift, nparents):
     .. warning:: This means negative indexing should **never**
                  be used with these arrays.
     """
+    if not list(upshift):
```
Comment (orbeckst) [inline package/MDAnalysis/core/topology.py]: casting to a `list` might be a performance issue for large arrays. Perhaps check explicitly for `None`, `len()` and whatever else is needed. Or use a `try/except`.

File: package/MDAnalysis/core/universe.py [hunk 2]
```diff
@@ -464,14 +464,18 @@ def empty(cls, n_atoms, n_residues=None, n_segments=None,
         .. versionchanged:: 0.19.0
            The attached Reader when trajectory=True is now a MemoryReader
         """
-        if n_residues is None:
```
Comment (orbeckst) [inline package/MDAnalysis/core/universe.py]: Add a `.. versionchanged:: 0.21.0` entry.

File: testsuite/MDAnalysisTests/core/test_universe.py [hunk 3]
```diff
@@ -765,3 +765,9 @@ def test_forces(self):
 
         assert u.atoms.positions.shape == (10, 3)
         assert u.atoms.forces.shape == (10, 3)
+
+    def test_empty_no_atoms(self):
+        u = mda.Universe.empty(0)
+        assert len(u.atoms) == 0
+        assert len(u.residues) == 0
+        assert len(u.segments) == 0
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/core/test_universe.py]: add newline at end of file – I am pretty sure that PEP8 or pylint will yell.

File: package/MDAnalysis/core/topology.py (hunk 1 above)
Comment (lilyminium) [reply package/MDAnalysis/core/topology.py]: Switched to checking the length of the list/array.

**Comment by orbeckst** [comment]
@richardjgowers I am assigning this to you for merging – if you're happy just merge, if there's anything else that I am not seeing, block the merge. Thanks.

**Comment by richardjgowers** [comment]
@orbeckst there was a tiny typo in the changelog, so I think that just became my 1,000th commit :laughing:

**Comment by orbeckst** [comment]
How fitting!

**Pull request merged**